File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.py
class Solution:
    def largestInteger(self, nums: List[int], k: int) -> int:
        subarray_count=[0]*(51)
        freq=[0]*(51)
        i=0
        j=0
        maxi=-1
        n=len(nums)
        while(j<n):
            freq[nums[j]]+=1
            if(j-i+1==k):
                for c in range(0,51):
                    if(freq[c]>0):
                        subarray_count[c]+=1
                freq[nums[i]]-=1
                i+=1
            j+=1
        for d in range(0,51):
            if(subarray_count[d]==1):
                maxi=max(maxi,d)
        return maxi

File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.java
class Solution {
    public int largestInteger(int[] nums, int k) {
        int[] subarray_count=new int[51];
        int[] freq=new int[51];
        int i=0;
        int j=0;
        int maxi=-1;
        while(j<nums.length)
        {
            freq[nums[j]]++;
            if(j-i+1==k)
            {
                for(int c=0;c<51;c++)
                {
                    if(freq[c]>0)
                    {
                        subarray_count[c]++;
                    }
                }
                freq[nums[i]]--;
                i++;
            }
            j++;
        }
        for(int d=0;d<51;d++)
        {
            if(subarray_count[d]==1)
            {
                maxi=Math.max(maxi,d);
            }
        }
        return maxi;
    }
}